pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1931,6 +1950,72 @@ SET search_path = pg_temp;
 REVOKE ALL ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) FROM PUBLIC;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.register_metric_view(schema_name name, view_name name)
+    RETURNS BOOLEAN
+AS $func$
+DECLARE
+   metric_table_name name;
+BEGIN
+    -- check if table/view exists
+    PERFORM * FROM information_schema.tables 
+    WHERE  table_schema = register_metric_view.schema_name
+    AND    table_name   = register_metric_view.view_name;
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'cannot register non-existant metric view in specified schema';
+    END IF;
+
+    -- cannot register view in data schema
+    IF schema_name = 'SCHEMA_DATA' THEN
+        RAISE EXCEPTION 'cannot register metric view in data schema';
+    END IF; 
+
+    -- check if view is based on a metric from prom_data
+    SELECT table_name FROM information_schema.view_table_usage
+    INTO metric_table_name
+    WHERE view_schema = register_metric_view.schema_name
+    AND view_table_usage.view_name = register_metric_view.view_name
+    AND table_schema = 'SCHEMA_DATA';
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'view not based on a metric table from SCHEMA_DATA schema';
+    END IF;
+
+    -- check if the view columns match columns from the underlying metric table
+    PERFORM m.column_name 
```

Comment:
hmm maybe it's better to check 3 things:
- there is a column called series_id (type bigint)
- there is a column called time (type timestamptz)
- there is a column called value (type float)

I don't think we really care if there are other columns or anything and I think this check would be more forward compatible.

Reply:
OK, will change that.